zcash_client_backend/src/data_api.rs

```diff
@@ -2848,6 +2848,11 @@ pub trait WalletWrite: WalletRead {
         received_tx: DecryptedTransaction<Self::AccountId>,
     ) -> Result<(), Self::Error>;
 
+    /// Marks the given transaction as being trusted, such that its outputs will be available for
+    /// spending with [`ConfirmationsPolicy::trusted`] confirmations even if the output is not
+    /// wallet-internal.
+    fn mark_tx_trusted(&mut self, txid: TxId) -> Result<(), Self::Error>;
```

Comment:
Design question: do we want to allow unmarking txs that were previously marked? I'm thinking of the "fat-finger" use case, where a user accidentally presses a button they didn't mean to, and wants to undo the change. This could be solved-ish in UIs by delaying application of the button (like how "Undo" buttons are implemented on emails), but that's a bit weird here as it's entirely local.

Reply:
Yeah, that's probably valuable.